src/mlpack/tests/test_catch_tools.hpp

```diff
@@ -22,14 +22,21 @@
     REQUIRE(std::abs((R) - (L)) <= (E) * std::abs(R))
 
 // Check the values of two matrices.
-inline void CheckMatrices(const arma::mat& a,
-                          const arma::mat& b,
+template <typename MatTypeA, typename MatTypeB,
+    typename = std::enable_if_t<arma::is_arma_type<MatTypeA>::value
+        && arma::is_arma_type<MatTypeB>::value
+        && !std::is_integral<typename MatTypeA::elem_type>::value
+        && !std::is_integral<typename MatTypeB::elem_type>::value>>
+inline void CheckMatrices(const MatTypeA& _a,
+                          const MatTypeB& _b,
```

Comment:
I think we can avoid this (and use of the internal Armadillo structures) by instead using the `.eval()` function at the call site, like this:

```
CheckMatrices((expression1 + expression2).eval(), mat2);
```

or something along those lines.  This will ensure that the type passed is `arma::mat` (or similar).  I actually can't find the place where `CheckMatrices()` is called with an expression, which would require the changes here (or the suggestion above)... can you point me to that?